### PR TITLE
End of life of Python 3.8

### PR DESCRIPTION
**Pull request by ryokan0123**
- Removed python 3.8 from allowed dependency
- Update the libraries
- Fix some tests

### CODE REVIEW

File: tests/dummy_modules/llama-seq-classification-tiny/tokenizer_config.json [hunk 1]
```diff
@@ -138,5 +138,6 @@
   "sep_token": "<sep>",
   "sp_model_kwargs": {},
   "tokenizer_class": "T5Tokenizer",
-  "unk_token": "<unk>"
+  "unk_token": "<unk>",
+  "chat_template": "{%- for message in messages %}\n    {%- if message['role'] == 'user' %}\n        {{- '<|user|>' + message['content'] + eos_token -}}\n    {%- elif message['role'] == 'system' %}\n        {{- '<|system|>' + message['content'] + eos_token -}}\n    {%- elif message['role'] == 'assistant' %}\n        {{- '<|assistant|>' + message['content'] + eos_token }}\n    {%- endif %}\n{%- if loop.last and add_generation_prompt -%}\n  {{- '<|assistant|>' -}}\n{%- endif -%}\n{%- endfor %}"
```
Comment (ryokan0123) [inline tests/dummy_modules/llama-seq-classification-tiny/tokenizer_config.json:142]: With latest `transformers`, we need to set chat_template to use it.
(Older versions automatically set the default template, but not now)

File: tests/core/language_model/test_hf_lm.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def test_batch_complete_text(lm: HuggingFaceLM) -> None:
     assert isinstance(completions[0], str)
 
 
-def test_compete_text(lm: HuggingFaceLM) -> None:
+def test_complete_text(lm: HuggingFaceLM) -> None:
```
Comment (ryokan0123) [inline tests/core/language_model/test_hf_lm.py:115]: fix typo

File: tests/core/language_model/test_hf_lm.py [hunk 3]
```diff
@@ -211,17 +211,22 @@ def test_if_random_seed_fixes_the_lm_outputs(lm_init_func: Callable[..., Hugging
...
 
-def test_batch_generate_chat_response(lm: LanguageModel) -> None:
-    responses = lm.batch_generate_chat_response([[{"role": "user", "content": "こんにちは。"}]], max_length=40)
+@pytest.fixture(scope="module")
+def chat_lm(model_name: str = "sbintuitions/tiny-lm-chat") -> HuggingFaceLM:
+    return HuggingFaceLM(model=model_name, model_kwargs={"torch_dtype": "float32"})
+
+
+def test_batch_generate_chat_response(chat_lm: LanguageModel) -> None:
+    responses = chat_lm.batch_generate_chat_response([[{"role": "user", "content": "こんにちは。"}]], max_length=40)
```
Comment (ryokan0123) [inline tests/core/language_model/test_hf_lm.py:220]: Tests failed because running `batch_generate_chat_response` with the model without a chat template.

**Review by butsugiri — APPROVED** [review]

File: tests/dummy_modules/llama-seq-classification-tiny/tokenizer_config.json (hunk 1 above)
Comment (Ktakuya332C) [reply tests/dummy_modules/llama-seq-classification-tiny/tokenizer_config.json:142]: [memo] chate_template of llama-seq-classification model is used in here
https://github.com/sbintuitions/flexeval/blob/aa1c87d6a94bcafc32e16842954cd60f3b5b93d8/flexeval/core/reward_model/sequence_classification.py#L39
while executing this test
https://github.com/sbintuitions/flexeval/blob/aa1c87d6a94bcafc32e16842954cd60f3b5b93d8/tests/core/reward_model/test_sequence_classification.py#L12

**Review by Ktakuya332C — APPROVED** [review]

**Pull request merged**